File: BOJ/itertools/combination.py
def comb(n, m):
    if n == m or m == 0:
        return 1
    return comb(n - 1, m - 1) + comb(n - 1, m)

def main():
    # nCr, n개중 r개를 뽑는다
    n, m = map(int,(input().split()))
    return comb(n+m,n)
# 8C2 == 8C6
if __name__ == '__main__':
    print(main())

File: Java/combination.java
class Solution {
    public int solution(int n, int m) {
        return comb(n + m, n);
    }
    private int comb(int n, int r) {
        if(r==0||n== r){
            return 1;
        }
        return comb(n - 1, r - 1) + comb(n - 1, r);
    }
}